common/configuration/1password_provider.go

```diff
@@ -61,3 +106,93 @@ func (o OnePasswordProvider) Store(ctx context.Context, ref Ref, value []byte) (
 }
 
 func (o OnePasswordProvider) Writer() bool { return o.OnePassword }
+
+type entry struct {
+	ID    string `json:"id"`
+	Value string `json:"value"`
+}
+
+type fullResponse struct {
+	Fields []entry `json:"fields"`
+}
+
+// Decode a full item response from op
+func decodeFull(output []byte) ([]entry, error) {
+	var full fullResponse
+	if err := json.Unmarshal(output, &full); err != nil {
+		return nil, fmt.Errorf("error decoding op full response: %w", err)
+	}
+	return full.Fields, nil
+}
+
+// Decode a single field from op
+func decodeSingle(output []byte) (*entry, error) {
+	var single entry
+	if err := json.Unmarshal(output, &single); err != nil {
+		return nil, fmt.Errorf("error decoding op single response: %w", err)
+	}
+	return &single, nil
+}
+
+// Custom parser for 1Password secret references because the format is not a standard URL, and we also need to
+// allow users to omit the field name so that we can support secrets with multiple fields.
+//
+// Does not support "section-name".
+//
+//	op://<vault-name>/<item-name>[/<field-name>]
+//
+// Secret references are case-insensitive and support the following characters:
+//
+//	alphanumeric characters (a-z, A-Z, 0-9), -, _, . and the whitespace character
+//
+// If an item or field name includes a / or an unsupported character, use the item
+// or field's unique identifier (ID) instead of its name.
+//
+// See https://developer.1password.com/docs/cli/secrets-reference-syntax/
+type secretRef struct {
```

Comment:
Once the other change goes in where we completely manage the secrets, we could probably remove this and just implement our own URI scheme that is valid.

Reply:
Had the same thought.